### PR TITLE
MGMT-16601: allow installation without providing host ip in advance

### PR DESCRIPTION
**Pull request by tsorya**
[MGMT-16601](https://issues.redhat.com//browse/MGMT-16601): allow installation without providing host ip in advance
In case host ip was not provided in manifest.json post-pivot configuration will start manually nodeip service, wait till it creates primary-ip file and read ip from it.
After getting ip it will create dnsmasq and dispatcher script override file and restart NM with dnsmasq

### CODE REVIEW

File: ibu-imager/postpivot/postpivot.go [hunk 1]
```diff
@@ -51,6 +52,11 @@ func NewPostPivot(scheme *runtime.Scheme, log *logrus.Logger, ops ops.Ops,
 	}
 }
 
+const (
+	nodeIpFile       = "/run/nodeip-configuration/primary-ip"
+	dnsmasqOverrides = "/etc/default/sno_dnsmasq_configuration_overrides"
+)
+
 func (p *PostPivot) PostPivotConfiguration(ctx context.Context) error {
```
Comment (eranco74) [inline ibu-imager/postpivot/postpivot.go:60]: This gets executed before or after prepare-installation-configuration.sh?

**Review by eranco74 — COMMENTED** [review]
Overall this looks good!
Moving logic out of prepare-installation-configuration.sh script is great, I think that in the end it can just wait for the configuration, and configure the networking if network config was provided.
Please add UT...

File: ibu-imager/postpivot/postpivot.go (hunk 1 above)
Comment (tsorya) [reply ibu-imager/postpivot/postpivot.go:60]: Yep, it is installation-configuration.service and yes my idea to remove prepare-installation-configuration.sh at all at the end

**Pull request merged**